source/tyr/tests/integration/purge_test.py

```diff
@@ -0,0 +1,154 @@
+
+from tyr import app, tasks
+from navitiacommon import models
+from tests.check_utils import api_get
+try:
+    import ConfigParser
+except:
+    import configparser as ConfigParser
+
+
+def create_dataset(dataset_type, dir):
+    dataset = models.DataSet()
+    dataset.type = dataset_type
+    dataset.family_type = '{}_family'.format(dataset_type)
+    dataset_backup_dir = tempfile.mkdtemp(dir=dir)
+    dataset.name = dataset_backup_dir
+    models.db.session.add(dataset)
+
+    metric = models.Metric()
+    metric.type = '{}2ed'.format(dataset_type)
+    metric.dataset = dataset
+    models.db.session.add(metric)
+
+    return dataset, metric
+
+
+def create_job(creation_date, dataset_type, backup_dir):
+    job = models.Job()
+    job.state = 'done'
+    dataset, metric = create_dataset(dataset_type, backup_dir)
+    job.data_sets.append(dataset)
+    job.metrics.append(metric)
+    job.created_at = creation_date
+    models.db.session.add(job)
+
+    return job
+
+
+def create_instance(name, jobs):
+    instance = models.Instance(name=name, jobs=jobs)
+    models.db.session.add(instance)
+    models.db.session.commit()
+
+
+def create_instance_with_same_type_datasets(name, backup_dir):
+    with app.app_context():
+        job_list = []
+        for index in range(3):
+            job_list.append(create_job(datetime.utcnow() - timedelta(days=index), 'poi', backup_dir))
+        create_instance(name, job_list)
+
+
+def create_instance_with_one_type_dataset(name, backup_dir):
+    with app.app_context():
+        job_list = []
+        for index, dataset_type in enumerate(['fusio', 'osm', 'poi']):
+            job_list.append(create_job(datetime.utcnow()-timedelta(days=index), dataset_type, backup_dir))
+        create_instance(name, job_list)
+
+
+def init_test():
+    config = ConfigParser.ConfigParser()
+
+    files = os.listdir(app.config['INSTANCES_DIR'])
+    for index, f in enumerate(files):
+        if os.path.splitext(f)[1] == '.ini':
+            instance_name = os.path.splitext(f)[0]
+            config.read(os.path.realpath('{}/{}'.format(app.config['INSTANCES_DIR'], f)))
+            backup_dir = config.get('instance', 'backup-directory')
+
+    return instance_name, backup_dir
+
+
+@pytest.mark.usefixtures("init_instances_dir")
+def test_purge_old_jobs():
+    """
+    Delete old jobs created before the time limit
+    """
+    app.config['JOB_MAX_PERIOD_TO_KEEP'] = 1
+
+    instance_name, backup_dir = init_test()
+    create_instance_with_same_type_datasets(instance_name, backup_dir)
+
+    instances_resp = api_get('/v0/instances')
+    assert len(instances_resp) == 1
+
+    jobs_resp = api_get('/v0/jobs/{}'.format(instances_resp[0]['name']))
+    assert len(jobs_resp['jobs']) == 3
+
+    tasks.purge_jobs()
+
+    jobs_resp = api_get('/v0/jobs/{}'.format(instances_resp[0]['name']))
+    assert len(jobs_resp['jobs']) == 1
+
+    folders = set(glob.glob('{}/*'.format(backup_dir)))
+    assert len(folders) == 1
```

Comment:
You probably need a `teardown_test` method to cleanup the files (and maybe in db).
And you need to use it in both tests.

Reply:
All the temp folders are deleted in the fixture init_instances_dir() called by both tests
-> shutil.rmtree(instance_dir).
The folders aren't present after the tests execution, verified.
the db is also cleaned by an already existing fixture clean_db()

Reply:
Ah yes, nicely handled, my bad.